Sitefinity CLI/Services/ProjectService.cs

```diff
@@ -0,0 +1,164 @@
+﻿using Newtonsoft.Json.Linq;
+using Newtonsoft.Json;
+using Sitefinity_CLI.Exceptions;
+using Sitefinity_CLI.Services.Interfaces;
+using Sitefinity_CLI.VisualStudio;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Sitefinity_CLI.Model;
+using Microsoft.Extensions.Logging;
+
+namespace Sitefinity_CLI.Services
+{
+    internal class ProjectService : IProjectService
+    {
+        public ProjectService(IHttpClientFactory clientFactory, IProjectConfigFileEditor projectConfigFileEditor, ICsProjectFileEditor csProjectFileEditor, ILogger<ProjectService> logger)
+        {
+            this.httpClient = clientFactory.CreateClient();
+            this.projectConfigFileEditor = projectConfigFileEditor;
+            this.csProjectFileEditor = csProjectFileEditor;
+            this.logger = logger;
+        }
+
+        public async Task<string> GetLatestSitefinityVersion()
+        {
+            using HttpRequestMessage request = new(HttpMethod.Get, Constants.SfAllNugetUrl);
+            HttpResponseMessage response = this.httpClient.Send(request);
+            string contentString = await response.Content.ReadAsStringAsync();
+            object jsonContent = JsonConvert.DeserializeObject(contentString);
+            JObject firstEntry = (jsonContent as JArray).First as JObject;
+            string latestVersion = (firstEntry["LatestVersion"]["Version"] as JValue).Value as string;
+
+            if (string.IsNullOrEmpty(latestVersion))
+            {
+                throw new ArgumentException("Can't get the latest Sitefinity version. Please specify the upgrade version.");
+            }
+
+            this.logger.LogInformation(string.Format(Constants.LatestVersionFound, latestVersion));
+            return latestVersion;
+        }
+
+        public IDictionary<string, string> GetConfigsForProjectsWithoutSitefinity(IEnumerable<string> projectsWithouthSfreferencePaths)
+        {
+            var configsWithoutSitefinity = new Dictionary<string, string>();
+            foreach (var project in projectsWithouthSfreferencePaths)
+            {
+                var porjectConfigPath = this.projectConfigFileEditor.GetProjectConfigPath(project);
+                if (!string.IsNullOrEmpty(porjectConfigPath))
+                {
+                    var configValue = File.ReadAllText(porjectConfigPath);
+                    configsWithoutSitefinity.Add(porjectConfigPath, configValue);
+                }
+            }
+
+            return configsWithoutSitefinity;
+        }
+
+        public IEnumerable<string> GetProjectPathsFromSolution(string solutionPath, string version, bool onlySitefinityProjects = false)
```

Comment:
The version property is used only when the optional parameter is true. If it is false it would not matter what is the value of the version property.
Maybe we can split it into two methods, One that retrieves all projects form the solution and one that retreives only sitefinity projects. The second one can have the version parameter. Also if there is a common logic between the methods we can reuse it (make a private method or something like that)